compiler/apps/playground/components/Editor/EditorImpl.tsx

```diff
@@ -98,7 +114,46 @@ function parseFunctions(
       suggestions: null,
     });
   }
-  return items;
+
+  return items.map<{
+    compile: boolean;
+    fn: FunctionLike;
+  }>(fn => {
+    const {body} = fn.node;
+    if (t.isBlockStatement(body)) {
+      const selfCheck = checkExplicitMemoizeDirectives(body.directives);
+      if (selfCheck === MemoizeDirectiveState.Enabled)
+        return {compile: true, fn};
+      if (selfCheck === MemoizeDirectiveState.Disabled)
+        return {compile: false, fn};
+
+      const parentWithDirective = fn.findParent(parentPath => {
+        if (parentPath.isBlockStatement() || parentPath.isProgram()) {
+          const directiveCheck = checkExplicitMemoizeDirectives(
+            parentPath.node.directives,
+          );
+          return [
+            MemoizeDirectiveState.Enabled,
+            MemoizeDirectiveState.Disabled,
+          ].includes(directiveCheck);
+        }
+        return false;
+      });
+
+      if (!parentWithDirective) return {compile: true, fn};
+      const parentDirectiveCheck = checkExplicitMemoizeDirectives(
+        (parentWithDirective.node as t.Program | t.BlockStatement).directives,
+      );
+      return {
+        compile: [
+          MemoizeDirectiveState.Enabled,
+          MemoizeDirectiveState.Undefined,
+        ].includes(parentDirectiveCheck),
+        fn,
+      };
+    }
+    return {compile: false, fn};
+  });
```

Comment:
rather than map over this list again, it might be more straightforward to do this in the traversal above and change the type of `items` to be your new return type

Reply:
Good point @poteto . changed it